packages/amplify_test/lib/src/integration_test_utils/auth_cognito/integration_test_auth_utils.dart

```diff
@@ -156,7 +156,10 @@ Future<void> adminCreateUser(
 
 /// Returns the OTP code for [username]. Must be called before the network call
 /// generating the OTP code.
-Future<String> getOtpCode(String username) async {
+Future<String> getOtpCode(
```

Comment:
There seems to be a bit of duplication everywhere this is consumed (creating the completer, passing it in, awaiting the future). Is there a reason to pass in `onEstablished` or could this be done internally?

You could change the response to `Future<OtpResult>` where `OtpResult` contains a member `Future<String> code`.

Instead of:
```dart
final establishedCompleter = Completer<void>();
final code = getOtpCode(
  username,
  onEstablished: establishedCompleter.complete,
);
await establishedCompleter.future;
...
await code
```

Consuming this would look like:
```dart
final otpResult = await getOtpCode(username);
...
await otpResult.code;
```

The implementation would look something like this. I haven't tested this, but I believe this would work.
```dart
class OtpResult {
  OtpResult({required this.code});
  Future<String> code;
}

/// Returns a result, which containts a future for getting the OTP code for [username].
///
/// The [OtpResult] future should complete before the network call
/// generating the OTP code is made.
Future<OtpResult> getOtpCode(String username) async {
  const subscriptionDocument = r'''
    subscription OnCreateMFACode($username: String!) {
      onCreateMFACode(username: $username) {
        username
        code
      }
    }''';

  final Completer<void> established = Completer<void>();

  final Stream<GraphQLResponse<String>> operation = Amplify.API.subscribe(
    GraphQLRequest<String>(
      document: subscriptionDocument,
      variables: {
        'username': username,
      },
    ),
    onEstablished: () {
      _logger.debug('Established connection');
      established.complete();
    },
  );

  final code = operation
      .tap(
        (event) => _logger.debug(
          'Got event: ${event.data}, errors: ${event.errors}',
        ),
      )
      .map((event) {
        if (event.hasErrors) {
          throw Exception(event.errors);
        }
        final json = jsonDecode(event.data!)['onCreateMFACode'] as Map;
        return CreateMFACodeResponse.fromJson(json.cast());
      })
      .map((event) => event.code)
      .first;

  await established.future;

  return OtpResult(code: code);
}
```

Reply:
Good idea

Reply:
agree this is nice, added this